src/metabase/public_settings.clj

```diff
@@ -71,6 +71,25 @@
   :audit   :getter
   :default true)
 
+(defn- set-update-channel! [new-channel]
+  (let [valid-channels ["latest" "beta" "nightly"]]
+    ;; check to make sure `update-channel` is a valid channel, or throw an Exception it is it not.
+    (when-not (some #{new-channel} valid-channels)
+      (throw (IllegalArgumentException.
+              (tru "Invalid update channel ''{0}''. Valid channels are: {1}"
+                   new-channel valid-channels))))
+    (setting/set-value-of-type! :string :update-channel new-channel)))
+
+(defsetting update-channel
+  (deferred-tru "Select an update channel to customize which Metabase releases you receive notifications for.")
+  :visibility :admin
+  :type       :string
+  :encryption :no
+  :export?    true
+  :audit      :getter
+  :setter     set-update-channel!
+  :default    "latest")
```

Comment:
@metabase/core-backend-admin-webapp please look at this extra closely - this is not my native tongue

Reply:
Pop this into the bottom of `test/metabase/public_settings_test.clj`

```
(deftest update-channel-test
  (testing "we can set the update channel"
    (mt/discard-setting-changes [update-channel]
      (public-settings/update-channel! "nightly")
      (is (= "nightly" (public-settings/update-channel)))))
  (testing "we can't set the update channel to an invalid value"
    (mt/discard-setting-changes [update-channel]
      (is (thrown?
           IllegalArgumentException
           (public-settings/update-channel! "millennially"))))))
```